projects/solar/test/test_mppt.c

```diff
@@ -0,0 +1,84 @@
+#include "log.h"
+#include "mppt.h"
+#include "spv1020_mppt.h"
+#include "spv1020_mppt_defs.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_SPI_PORT SPI_PORT_2
+
+void setup_test() {
+  gpio_init();
+  SpiSettings spi_settings = { .baudrate = 60000,
+                               .mode = SPI_MODE_3,
+                               .mosi = { .port = GPIO_PORT_B, 15 },
+                               .miso = { .port = GPIO_PORT_B, 14 },
+                               .sclk = { .port = GPIO_PORT_B, 13 },
+                               .cs = { .port = GPIO_PORT_B, 12 } };
+  spi_init(TEST_SPI_PORT, &spi_settings);
+}
+
+void teardown_test() {}
+
+// Test that we can successfully send the SHUT and Turn ON commands.
+void test_shut_and_turn_on(void) {
```

Comment:
The tests for the range of the `pin` parameter are good :) To test more `mppt`-specific functionality, maybe you could mock `mux_set` (see `test_sense.c` for how to mock a function) and make sure that it's called correctly on each `mppt` function call?